tests/rados/test_rados_watcher_cleanup.py

```diff
@@ -0,0 +1,564 @@
+            )
+            _final_cleanup_mapped_devices(client_nodes, pool, rados_obj, mapped_devices)
+            _final_cleanup_rbd_images(pool, rados_obj, all_created_images)
+            log.info("Attempting to delete pool: %s.", pool)
+            if not rados_obj.delete_pool(pool=pool):
+                log.error(f"Failed to delete pool {pool} during final cleanup.")
+        except Exception as e:
+            log.error("Critical error during final cleanup of pool: %s", e)
+            log.exception(e)
+        if rados_obj.check_crash_status():
+            log.error(
+                "Test finished, but crash detected during or after test execution."
+            )
+
+
+def _verify_watcher_presence(image_name, client, pool, rados_obj, client_ip):
+    log.info("Verifying watcher created for %s from %s", image_name, client_ip)
+    max_retries = 10
+    for i in range(max_retries):
+        watchers = rados_obj.get_rbd_client_ips(pool, image_name)
+        if watchers and client_ip in watchers:
+            log.info(
+                "Watcher successfully created for %s by %s.", image_name, client_ip
+            )
+            return client_ip
+        log.debug(
+            "Watcher not yet seen for %s by %s. Retrying... (Attempt %d)",
+            image_name,
+            client_ip,
+            i + 1,
+        )
+        time.sleep(2)
+    raise Exception(
+        "Watcher for %s not found after mapping from %s. Current watchers: %s"
+        % (image_name, client_ip, watchers)
+    )
+
+
+def _force_cleanup_device(client, device, image_name):
+    if not device:
+        log.debug(
+            "No device provided for image %s. Skipping forceful cleanup.", image_name
+        )
+        return
+    mount_path = "/mnt/%s" % image_name
+    log.info("Attempting to unmount %s if mounted.", mount_path)
+    client.exec_command(cmd="umount %s" % mount_path, sudo=True, check_ec=False)
+    time.sleep(1)
+    log.info("Checking for processes holding %s.", device)
+    pids_out, _ = client.exec_command(
+        cmd="lsof -t %s" % device, sudo=True, check_ec=False
+    )
+    pids = [p.strip() for p in pids_out.splitlines() if p.strip()]
+    if pids:
+        log.warning(
+            "Found PIDs holding %s: %s. Attempting to kill them.",
+            device,
+            ", ".join(pids),
+        )
+        for pid in pids:
+            client.exec_command(cmd="kill -9 %s" % pid, sudo=True, check_ec=False)
+        time.sleep(5)
+        pids_after_kill_out, _ = client.exec_command(
+            cmd="lsof -t %s" % device, sudo=True, check_ec=False
+        )
+        if [p.strip() for p in pids_after_kill_out.splitlines() if p.strip()]:
+            log.error("Critical: PIDs still holding %s after kill -9.", device)
+    else:
+        log.info("No specific processes found holding %s.", device)
+    log.info("Attempting to pkill common RBD client processes.")
+    client.exec_command(cmd="pkill -9 rbd-nbd", sudo=True, check_ec=False)
```

Comment:
I think just umount of the mount path should be enough for clean up on client node. no need to kill the services. 

as long as that mount is no longer on client, we can delete the mount path created. that's enough.